synthesis/synthesis.go

```diff
@@ -342,78 +248,67 @@ func FixCds(sqlitePath string, sequence string, codontable codon.Table, problema
 			if !validBias {
 				return sequence, []Change{}, fmt.Errorf("Invalid bias. Expected NA, GC, or AT, got %s", suggestion.Bias)
 			}
-			// First, let's insert the suggestions that we found using our problematicSequenceFuncs
-			_, err := db.Exec(`INSERT INTO suggestedfix(step, start, end, gcbias, quantityfixes, suggestiontype) VALUES (?, ?, ?, ?, ?, ?)`, i, suggestion.Start, suggestion.End, suggestion.Bias, suggestion.QuantityFixes, suggestion.SuggestionType)
-			if err != nil {
-				return sequence, []Change{}, fmt.Errorf("Failed to insert suggestedfix (does start+end point to valid parts of the amino acid sequence?). Got SQL error: %s", err)
+
+			// For each suggestion, get a list of potential changes that could fix the problem.
+			var potentialChanges []Change
+			for positionSelector := suggestion.Start; positionSelector <= suggestion.End; positionSelector++ {
+				lastCodon := historicalMap[positionSelector][len(historicalMap[positionSelector])-1]
```

Comment:
Messy inline arithmetic. Expand.

Reply:
It's not even arithmetic, It's just selecting the last element of a list